benchmarks/profile.py

```diff
@@ -0,0 +1,228 @@
+import json
+import os
+import pathlib
+import re
+import sys
+import warnings
+
+import jax.numpy as jnp
+import matplotlib.pyplot as plt  # pyright: ignore[reportMissingImports]
+import numpy as np  # pyright: ignore[reportMissingImports]
+import pandas as pd  # pyright: ignore[reportMissingImports]
+
+
+def find_benchmark_directory():
+    """Find the .benchmarks directory in the current or parent directory."""
+    script_dir = pathlib.Path(__file__).parent
+    benchmarks_dir = script_dir / ".benchmarks"
+
+    if not benchmarks_dir.exists():
+        benchmarks_dir = script_dir.parent / ".benchmarks"
+
+    if not benchmarks_dir.exists():
+        raise FileNotFoundError(
+            "Error: Could not find .benchmarks directory. "
+            "Please ensure it exists in the current or parent directory."
+        )
+
+    print(f"Using benchmarks from: {benchmarks_dir}")
+    return benchmarks_dir
+
+
+def find_benchmark_run(benchmarks_dir: pathlib.Path, args: list) -> dict:
+    """Find the benchmark run data based on the provided arguments."""
+    platform, python_version, precision, iD = args[-4:]
```

Comment:
Maybe we should use https://github.com/google/python-fire to make the argument-parsing nice?

I think it's okay to introduce the extra dependency as we can put it in the dev dependency list.